### PR TITLE
voice messages: add support from bridging both directions

### PR DESCRIPTION
**Pull request by sumnerevans**
iMessage -> Matrix works correctly right now
Matrix -> iMessage still needs updates to barcelona to work, but this should have all of the mautrix infra necessary for it to work

Needs https://github.com/mautrix/go/pull/61

### CODE REVIEW

File: imessage/mac/send.go [hunk 1]
```diff
@@ -182,25 +180,21 @@ func (mac *macOSDatabase) SendMessage(chatID, text string, replyTo string, reply
...
-	if err != nil {
-		return nil, fmt.Errorf("failed to create temp dir: %w", err)
-	}
-	filePath := filepath.Join(dir, filename)
-	err = ioutil.WriteFile(filePath, data, 0640)
-	if err != nil {
-		return nil, fmt.Errorf("failed to write data to temp file: %w", err)
+func (mac *macOSDatabase) SendFile(chatID, filename string, pathOnDisk string, replyTo string, replyToPart int, mimeType string, voiceMemo bool) (*imessage.SendResponse, error) {
+	if voiceMemo {
+		return nil, fmt.Errorf("Voice memos are not supported")
```
Comment (EricRabil) [inline imessage/mac/send.go:185]: Should this actively fail or should we just send it as a regular attachment?

**Review by EricRabil — CHANGES_REQUESTED** [review]

**Review by EricRabil — APPROVED** [review]

File: portal.go [hunk 2]
```diff
@@ -1042,17 +1078,32 @@ func (portal *Portal) handleIMMemberChange(msg *imessage.Message, dbMessage *dat
...
 		portal.log.Errorfln("Failed to read attachment in %s: %v", msg.GUID, err)
-		return nil, fmt.Errorf("failed to read attachment: %w", err)
+		return nil, nil, fmt.Errorf("failed to read attachment: %w", err)
+	}
+
+	mimeType := attach.GetMimeType()
+	fileName := attach.GetFileName()
+	extraContent := map[string]interface{}{}
+	if msg.IsAudioMessage {
+		data, err = ffmpeg.ConvertBytes(data, ".ogg", []string{}, []string{"-c:a", "libopus"}, ".caf")
```
Comment (tulir) [inline portal.go:1092]: Shouldn't the last parameter be a mime type?

Comment (sumnerevans) [reply portal.go:1092]: https://github.com/mautrix/go/pull/61

Comment (sumnerevans) [reply portal.go]: It was working with this incorrectly set so :shrug:

**Review by tulir — APPROVED** [review]

**Pull request merged**